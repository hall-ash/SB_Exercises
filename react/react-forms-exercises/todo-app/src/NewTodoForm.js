import React, {useState} from 'react';

const NewTodoForm = ({addTodo}) => {
  const [todoInput, setTodoInput] = useState('');

  const handleChange = evt => {
    setTodoInput(evt.target.value);
  }

  const handleSubmit = evt => {
    evt.preventDefault();
    addTodo(todoInput);
    setTodoInput('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="todo">Todo:</label>
      <input type="text" id="todo" name="todo" onChange={handleChange} value={todoInput}/><br></br>
      <button>Add</button>
    </form>
  )

}

export default NewTodoForm;