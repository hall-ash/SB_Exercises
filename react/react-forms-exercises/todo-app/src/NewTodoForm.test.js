import React from 'react';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders without crashing', () => {
  render(<NewTodoForm />);
});

test('it matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});