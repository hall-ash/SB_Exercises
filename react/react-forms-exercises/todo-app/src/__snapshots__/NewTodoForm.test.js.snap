// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo:
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <br />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
